### ISSUE TITLE
Canada Post PWS (2) - parse_rates_response with a contract_id cause a REXML::ParseException  

### ISSUE THREAD

**Issue by dlepage** [Shopify/active_shipping#70]
XML response from CP contains '& amp;' and doesn't work well with the REXML::Text::unnormalize(response). 

Quick and dirty fix :
line : parse_rates_response(response, origin, destination)
replaced by:  parse_rates_response(response.gsub('& amp;','-'), origin, destination)
( or doc = REXML::Document.new(REXML::Text::unnormalize(response.gsub('& amp;','-')))

NOTE : remove space between & and amp

**Comment by mstate** [comment]
Just wanted to confirm that CP was rendering all the ampersands in their escaped form &amp; .

The call REXML::Text::unnormalize(response) turns the &amp; back into an & . 

Then, when it tries to parse this unnormalized response
`REXML::Document.new(REXML::Text::unnormalize(response))`

It throws the exeption:
`RuntimeError: Illegal character '&' in raw string "https://soa-gw.canadapost.ca/rs/ship/service/DOM.EP?contract=XXXXXXXXXX&country=CA"`

Here is a full response:
```
<?xml version='1.0' encoding='UTF-8'?>\n
<price-quotes xmlns='http://www.canadapost.ca/ws/ship/rate'>
    <price-quote>
        <service-code>DOM.EP</service-code>
        <service-link href='https://soa-gw.canadapost.ca/rs/ship/service/DOM.EP?contract=XXXXXXXXXX&amp;country=CA' media-type='application/vnd.cpc.ship.rate+xml' rel='service'/>
        <service-name>Expedited Parcel</service-name>
        <price-details>
            <base>5.07</base>
            <taxes>
                <gst>0.00</gst>
                <pst>0.00</pst>
                <hst percent='13.000'>0.68</hst>
            </taxes>
            <due>5.89</due>
            <options>
                <option>
                    <option-code>DC</option-code>
                    <option-name>Delivery confirmation</option-name>
                    <option-price>0</option-price>
                </option>
            </options>
            <adjustments>
                <adjustment>
                    <adjustment-code>FUELSC</adjustment-code>
                    <adjustment-name>Fuel surcharge</adjustment-name>
                    <adjustment-cost>0.14</adjustment-cost>
                    <qualifier>
                        <percent>2.75</percent>
                    </qualifier>
                </adjustment>
            </adjustments>
        </price-details>
        <weight-details/>
        <service-standard>
            <am-delivery>false</am-delivery>
            <guaranteed-delivery>true</guaranteed-delivery>
            <expected-transit-time>1</expected-transit-time>
            <expected-delivery-date>2015-02-13</expected-delivery-date>
        </service-standard>
    </price-quote>
    <price-quote>
        <service-code>DOM.PC</service-code>
        <service-link href='https://soa-gw.canadapost.ca/rs/ship/service/DOM.PC?contract=XXXXXXXXXX&amp;country=CA' media-type='application/vnd.cpc.ship.rate+xml' rel='service'/>
        <service-name>Priority</service-name>
        <price-details>
            <base>8.33</base>
            <taxes>
                <gst>0.00</gst>
                <pst>0.00</pst>
                <hst percent='13.000'>1.17</hst>
            </taxes>
            <due>10.15</due>
            <options>
                <option>
                    <option-code>DC</option-code>
                    <option-name>Delivery confirmation</option-name>
                    <option-price>0</option-price>
                </option>
            </options>
            <adjustments>
                <adjustment>
                    <adjustment-code>FUELSC</adjustment-code>
                    <adjustment-name>Fuel surcharge</adjustment-name>
                    <adjustment-cost>0.65</adjustment-cost>
                    <qualifier>
                        <percent>7.75</percent>
                    </qualifier>
                </adjustment>
            </adjustments>
        </price-details>
        <weight-details/>
        <service-standard>
            <am-delivery>false</am-delivery>
            <guaranteed-delivery>true</guaranteed-delivery>
            <expected-transit-time>1</expected-transit-time>
            <expected-delivery-date>2015-02-13</expected-delivery-date>
        </service-standard>
    </price-quote>
    <price-quote>
        <service-code>DOM.RP</service-code>
        <service-link href='https://soa-gw.canadapost.ca/rs/ship/service/DOM.RP?contract=XXXXXXXXXX&amp;country=CA' media-type='application/vnd.cpc.ship.rate+xml' rel='service'/>
        <service-name>Regular Parcel</service-name>
        <price-details>
            <base>5.07</base>
            <taxes>
                <gst>0.00</gst>
                <pst>0.00</pst>
                <hst percent='13.000'>0.68</hst>
            </taxes>
            <due>5.89</due>
            <options>
                <option>
                    <option-code>DC</option-code>
                    <option-name>Delivery confirmation</option-name>
                    <option-price>0</option-price>
                    <qualifier>
                        <included>true</included>
                    </qualifier>
                </option>
            </options>
            <adjustments>
                <adjustment>
                    <adjustment-code>FUELSC</adjustment-code>
                    <adjustment-name>Fuel surcharge</adjustment-name>
                    <adjustment-cost>0.14</adjustment-cost>
                    <qualifier>
                        <percent>2.75</percent>
                    </qualifier>
                </adjustment>
            </adjustments>
        </price-details>
        <weight-details/>
        <service-standard>
            <am-delivery>false</am-delivery>
            <guaranteed-delivery>false</guaranteed-delivery>
            <expected-transit-time>2</expected-transit-time>
            <expected-delivery-date>2015-02-16</expected-delivery-date>
        </service-standard>
    </price-quote>
    <price-quote>
        <service-code>DOM.XP</service-code>
        <service-link href='https://soa-gw.canadapost.ca/rs/ship/service/DOM.XP?contract=XXXXXXXXXX&amp;country=CA' media-type='application/vnd.cpc.ship.rate+xml' rel='service'/>
        <service-name>Xpresspost</service-name>
        <price-details>
            <base>5.14</base>
            <taxes>
                <gst>0.00</gst>
                <pst>0.00</pst>
                <hst percent='13.000'>0.72</hst>
            </taxes>
            <due>6.26</due>
            <options>
                <option>
                    <option-code>DC</option-code>
                    <option-name>Delivery confirmation</option-name>
                    <option-price>0</option-price>
                </option>
            </options>
            <adjustments>
                <adjustment>
                    <adjustment-code>FUELSC</adjustment-code>
                    <adjustment-name>Fuel surcharge</adjustment-name>
                    <adjustment-cost>0.40</adjustment-cost>
                    <qualifier>
                        <percent>7.75</percent>
                    </qualifier>
                </adjustment>
            </adjustments>
        </price-details>
        <weight-details/>
        <service-standard>
            <am-delivery>false</am-delivery>
            <guaranteed-delivery>true</guaranteed-delivery>
            <expected-transit-time>1</expected-transit-time>
            <expected-delivery-date>2015-02-13</expected-delivery-date>
        </service-standard>
    </price-quote>
</price-quotes>
```

**Comment by wvanbergen** [comment]
What version are you using? This should be fixed in the latest 1.0
prerelease.
